lifecycle/lifecycle.go

```diff
@@ -62,6 +76,18 @@ const (
 	CloudEventV1Format
 )
 
+//lint:ignore U1000 #1768 support for external clients
+var formatToString = map[Format]string{
+	UnknownFormat:      "UnknownFormat",
+	ChoriaFormat:       "ChoriaFormat",
+	CloudEventV1Format: "CloudEventV1Format",
+}
+
+//lint:ignore U1000 #1768 support for external clients
+func (f Format) String() string {
+	return formatToString[f]
```

Comment:
Think we need to be a bit more careful here to also handle `""` for example, so maybe:

```
s, ok := formatToString[f]
if !ok { return fomatToString[UnknownFormat] }
return s
```

Format is probably ok but the Type one we should handle similar and perhaps change `_ = iota` to `Unknown Type = iota` then use that.

As is, when the type 0 for some weird reason we'd get an empty string

Sorry for not noticing that earlier

Reply:
with 0 never being used and not having any name before - i think we can just call 0 unknown.

I always (well I try) skip 0 in these iotas since just doing `x:=X{}` if that X type holds a key of this type it would be 0, so I want real values set on purpose.  But it means we need to handle 0 gracefully in these kinds of tings